### PR TITLE
fix(TDP-5518): loading while retrieving semantic types

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-5518

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

File: dataprep-webapp/src/app/components/datagrid/header/datagrid-header-controller.js [hunk 1]
```diff
@@ -89,10 +81,21 @@ export default function DatagridHeaderCtrl($q, $scope, state,
...
 					vm.transformationsRetrieveError = true;
+				})
+				.finally(() => {
+					vm.initTransformationsInProgress = false;
 				});
 		}
+		vm.fetchMatchingSemanticTypes();
 	};
 
+	vm.fetchMatchingSemanticTypes = () => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/datagrid/header/datagrid-header-controller.js]: There is no unit tests for it?

File: dataprep-webapp/src/app/components/datagrid/header/datagrid-header-controller.js [hunk 2]
```diff
@@ -89,10 +81,21 @@ export default function DatagridHeaderCtrl($q, $scope, state,
 				})
 				.catch(() => {
 					vm.transformationsRetrieveError = true;
+				})
+				.finally(() => {
+					vm.initTransformationsInProgress = false;
 				});
 		}
+		vm.fetchMatchingSemanticTypes();
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/datagrid/header/datagrid-header-controller.js]: Please add to your unit tests this call

Comment (charlesnguyen) [reply dataprep-webapp/src/app/components/datagrid/header/datagrid-header-controller.js]: There are tests but not "unit". Done

File: dataprep-webapp/src/app/components/datagrid/header/datagrid-header-controller.js (hunk 1 above)
Comment (charlesnguyen) [reply dataprep-webapp/src/app/components/datagrid/header/datagrid-header-controller.js]: There are tests but not "unit". Done

**Pull request merged**